src/main/java/org/broadinstitute/hellbender/utils/diffengine/DiffElement.java

```diff
@@ -0,0 +1,173 @@
+package org.broadinstitute.hellbender.utils.diffengine;
+
+import com.google.common.base.Strings;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Represents an element in the tree of differences.
+ */
+final class DiffElement {
+
+    /**
+     * Special element that is on top of the hierarchy and has no parent.
+     */
+    public static final DiffElement ROOT = new DiffElement();
+    public static final String ROOT_NAME= "ROOT";
+
+    private final String name;
+    private final DiffElement parent;
+    private final DiffValue value;
+
+    /**
+     * For ROOT only
+     */
+    private DiffElement() {
+        this.name = ROOT_NAME;
+        this.parent = null;
+        this.value = new DiffValue(this, ROOT_NAME);
+    }
+
+    public DiffElement(final String name, final DiffElement parent, final DiffValue value) {
+        if ( name.equals(ROOT_NAME) ) {
+            throw new IllegalArgumentException("Cannot use reserved name ROOT");
+        }
+        this.name = name;
+        this.parent = parent;
+        this.value = value;
+        this.value.setBinding(this);
+    }
+
+    /**
+     * Makes a new element from the given parent and unparsed tree.
```

Comment:
You deleted the warning from the old codebase that this method "Doesn't support full tree structure parsing". Why? Is that warning no longer applicable?